### PR TITLE
Revise OpenCADD-KLIFS manuscript (issues #133 and #134)

### PR DESCRIPTION
**Pull request by dominiquesydow**
## Description
Address manuscript issues raised in https://github.com/volkamerlab/opencadd/issues/133 and https://github.com/volkamerlab/opencadd/issues/134.

I am sorting the (very good!) raised points here in the following grouped into the corresponding sections - adding a few additional checks that came to my mind.

Check manuscript: https://whedon.theoj.org/

### 1. _Summary_ section
- [x] Check if this section can stay as it is after having updated the other sections. > Yes but moved the comment on OpenCADD-KLIFS being part of the larger OpenCADD package here.

### 2. _Statement of need_ section
- [x] Shorter and more concise! Move some of the text to Description/**Key Features**/System Overview/Usage section
- [x] Check again if target audience is highlighted enough

### 3. _State of the field_ section
- [x] Check if this section is already part of the JOSS author's guide; if not, raise an issue so that it can be added > Maybe add "State of the field" section [here](https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain)? > Not needed; section not explicitly necessary, state of the field simply needs to be discussed somewhere.
- [x] How was data accessed previously?
  - via KLIFS website: [manually](https://klifs.net/)
  - via [KLIFS KNIME nodes](https://github.com/3D-e-Chem/knime-klifs)
  - via [KLIFS API](https://klifs.net/swagger_v2/) : programmatically 
    - with e.g. `requests`
    - with `bravado` that can turn OpenAPI in Python module
- [x] How does OpenCADD-KLIFS compare to other commonly-used packages? How is it easier now?
  - Less technical to set up (only a few lines of code without the need to know how to set up a `bravado`-generated Python module from the KLIFS OpenAPI definitions)
  - Makes switching between local and remote access extremely easy (API and output the same!)
  - Output is ready to work with (OpenAPI results are formatted as ABC classes; always needs additional lines of code to bring results in a format that is useful)
- [x] Is this package the open-source version of proprietary software? No, the underlying infrastructure is not proprietary.
- [x] Any other packages out there that offer ways to query/analyse PDB structure data in a similar way? Double-check but not to my knowledge - KLIFS is integrating and annotating different kinase resources 
  - Structures from the PDB _annotated with the KLIFS pockets of 85 residues that are aligned across all kinase structures, including a split of PDB files by all chains/models_
  -  KLIFS interaction fingerprint (only available at KLIFS) investigating kinase-ligand interactions seen in experimental structures
  - Bioactivity data measured against kinases (fetched from ChEMBL)
  - Kinase inhibitor data (fetched from PKIDB)
  - All entry points (kinases, structures, ligands) have unique identifiers in KLIFS, which makes it super easy to fetch all relevant information for a query kinase, structure, or ligand!

### 4. _Key Features_ section
- [x] Move bullet points from _Statement of need_ here

### Typos
- [x] Incomplete reference to Raschka on line 46

### Documentation!
- [x] Sync new _Statement of need_ with documentation! (after approval from reviewers)

## Status
- [x] Ready to go

### CODE REVIEW

File: paper/paper.md [hunk 1]
```diff
@@ -23,34 +23,56 @@ bibliography: paper.bib
...
+
 OpenCADD-KLIFS is aimed at current and future users of the KLIFS database who seek to 
 integrate kinase resources into Python-based research projects.
-This module offers access to KLIFS data [@Kanev:2021] such as information about kinases, structures, ligands, 
-interaction fingerprints, and bioactivities. 
-KLIFS thereby focuses especially on the ATP binding site, defined as a set of 85 residues and aligned across all structures using a multiple sequence alignment (MSA) [@vanLinden:2014].
 With OpenCADD-KLIFS, KLIFS data can be queried either locally from a KLIFS download or remotely from the KLIFS webserver. 
-The presented module provides identical APIs for the remote and local queries for KLIFS data and streamlines all output into 
+The presented module provides identical APIs for the remote and local queries and streamlines all output into 
 standardized Pandas DataFrames [@pandas] to allow for easy and quick downstream data analyses (\autoref{fig:opencadd_klifs_toc}). This Pandas-focused setup is ideal to work with in Jupyter notebooks [@Kluyver:2016].
```
Comment (AndreaVolkamer) [inline paper/paper.md:43]:
```suggestion
standardized Pandas DataFrames [@pandas] to allow for easy and quick downstream data analyses (\autoref{fig:opencadd_klifs_toc}). This Pandas-focused setup is ideal if you work with Jupyter notebooks [@Kluyver:2016]. 
```

Comment (AndreaVolkamer) [inline paper/paper.md:43]: Just doublechecking if correct: the pandas reference reads like this 'The pandas development team, 2020'?

File: paper/paper.md [hunk 2]
```diff
@@ -23,34 +23,56 @@ bibliography: paper.bib
...
 With OpenCADD-KLIFS, KLIFS data can be queried either locally from a KLIFS download or remotely from the KLIFS webserver. 
-The presented module provides identical APIs for the remote and local queries for KLIFS data and streamlines all output into 
+The presented module provides identical APIs for the remote and local queries and streamlines all output into 
 standardized Pandas DataFrames [@pandas] to allow for easy and quick downstream data analyses (\autoref{fig:opencadd_klifs_toc}). This Pandas-focused setup is ideal to work with in Jupyter notebooks [@Kluyver:2016]. 
 
-[OpenCADD-KLIFS](https://opencadd.readthedocs.io/en/latest/databases_klifs.html) (``opencadd.databases.klifs``) is a part of the [OpenCADD](https://opencadd.readthedocs.io/) package, a collection of Python modules for structural cheminformatics.
-
 ![OpenCADD-KLIFS fetches KLIFS data [@Kanev:2021] offline from a KLIFS download or online from the KLIFS database and formats the output as user-friendly Pandas DataFrames [@pandas].\label{fig:opencadd_klifs_toc}](opencadd_klifs_toc.png)
```
Comment (AndreaVolkamer) [inline paper/paper.md:45]:
```suggestion
![OpenCADD-KLIFS fetches KLIFS data [@Kanev:2021] offline from a local KLIFS download or online from the KLIFS database and formats the output as user-friendly Pandas DataFrames [@pandas].\label{fig:opencadd_klifs_toc}](opencadd_klifs_toc.png)
```

File: paper/paper.md [hunk 3]
```diff
@@ -23,34 +23,56 @@ bibliography: paper.bib
...
+# State of the field
+
+The KLIFS database is unique in the structure-based kinase field in terms of integrating and annotating different data resources in a kinase- and pocket-focused manner. Kinases, structures, and ligands have unique identifiers in KLIFS, which makes it possible to fetch and filter cross-referenced information for a query kinase, structure, or ligand.
+
+- Kinase structures are fetched from the PDB, split by chains and alternate models, annotated with the KLIFS pocket of 85 residues, and aligned across the fully structurally covered kinome.
+- Kinase-ligand interactions seen in experimental structures are annotated for the 85 pocket residues in the form of the KLIFS interaction fingerprint (KLIFS IFP).
+- Bioactivity data measured against kinases are fetched from ChEMBL [@Mendez:2018] and linked to kinases, structures, and ligands available in KLIFS.
+- Kinase inhibitor metadata are fetched from the PKIDB [@Carles:2018] and linked to co-crystallized ligands available in KLIFS.
+
+The KLIFS data integrations and annotations can be accessed in different ways, which are all open-sourced:
```
Comment (AndreaVolkamer) [inline paper/paper.md:56]: open-sourced or open source?

File: paper/paper.md [hunk 4]
```diff
@@ -23,34 +23,56 @@ bibliography: paper.bib
...
+
+- Manually via the [KLIFS website](https://klifs.net/) interface: This mode is preferable when searching for information on a specific structure or smaller set of structures.
+- Automated via the [KLIFS KNIME](https://github.com/3D-e-Chem/knime-klifs) nodes [@McGuire:2017; @Kooistra:2018]: This mode is extremely useful if the users' projects are embedded in KNIME; programming is not needed.
```
Comment (AndreaVolkamer) [inline paper/paper.md:59]:
```suggestion
- Automated via the [KLIFS KNIME](https://github.com/3D-e-Chem/knime-klifs) nodes [@McGuire:2017; @Kooistra:2018]: This mode is extremely useful if the users' projects are embedded in KNIME workflows; programming is not needed.
```

File: paper/paper.md [hunk 5]
```diff
@@ -23,34 +23,56 @@ bibliography: paper.bib
...
+- Programmatically using the REST API and KLIFS OpenAPI specifications: This mode is needed for users who seek to perform larger scale queries or integrate different queries into programmatic workflows. In the following, we will discuss this mode in context of Python-based projects and explain how OpenCADD-KLIFS improves the user experience.
```
Comment (AndreaVolkamer) [inline paper/paper.md:60]:
```suggestion
- Programmatically using the REST API and KLIFS OpenAPI specifications: This mode is needed for users who seek to perform larger scale queries or to integrate different queries into programmatic workflows. In the following, we will discuss this mode in context of Python-based projects and explain how OpenCADD-KLIFS improves the user experience.
```

File: paper/paper.md [hunk 6]
```diff
@@ -23,34 +23,56 @@ bibliography: paper.bib
...
+
+The KLIFS database offers standardized URL schemes (REST API), which allows users to query data by defined URLs, using e.g. the Python package requests [@requests]. Instead of writing customized scripts to generate such KLIFS URLs, the KLIFS OpenAPI specifications &mdash; a document that defines the KLIFS REST API scheme &mdash; can be used to generate a Python client, using e.g. the Python package bravado [@bravado]. This client offers a Python API to send requests and receive responses.
+This setup is already extremely useful, however, it has a few drawbacks: the setup is technical, the output is not easily readable for humans and not ready for immediate down-stream integrations &mdash; requiring similar but not identical reformatting functions for different query results &mdash;, and switching from remote requests to local KLIFS download queries is not possible. Facilitating and streamlining these tasks is the purpose of OpenCADD-KLIFS as discussed in more detail in the next section.
```
Comment (AndreaVolkamer) [inline paper/paper.md:63]:
```suggestion
This setup is already extremely useful, however, it has a few drawbacks: the setup is highly technical, the output is not easily readable for humans and not ready for immediate down-stream integrations &mdash; requiring similar but not identical reformatting functions for different query results &mdash;, and switching from remote requests to local KLIFS download queries is not readily possible. Facilitating and streamlining these tasks is the purpose of OpenCADD-KLIFS as discussed in more detail in the next section.
```

Comment (AndreaVolkamer) [reply paper/paper.md:63]: above we say 'downstream' here 'down-stream', let's choose one.

File: paper/paper.md [hunk 7]
```diff
@@ -23,34 +23,56 @@ bibliography: paper.bib
...
+
+# Key Features
+
+The KLIFS database offers a REST API compliant with the OpenAPI specification [@klifsswagger]. Our module OpenCADD-KLIFS uses bravado to dynamically generate a Python client based on the OpenAPI definitions and adds wrappers to enable the following functionalities:
 
 - A session is set up, which allows access to various KLIFS *data sources* by different *identifiers* with the API ``session.data_source.by_identifier``. *Data sources* currently include kinases, structures and annotated conformations, modified residues, pockets, ligands, drugs, and bioactivities; *identifiers* refer to kinase names, PDB IDs, KLIFS IDs, and more. For example, ``session.structures.by_kinase_name`` fetches information on all structures for a query kinase.
```
Comment (AndreaVolkamer) [inline paper/paper.md:69]: Could we state: 'A session is set up automatically, ...'?

**Review by AndreaVolkamer — DISMISSED** [review]
@dominiquesydow great integration of the feedback, I only added minor textual edit suggestions.

File: paper/paper.md (hunk 1 above)
Comment (dominiquesydow) [reply paper/paper.md]: That is their official bibtex entry: https://pandas.pydata.org/about/citing.html

File: paper/paper.md (hunk 3 above)
Comment (dominiquesydow) [reply paper/paper.md:56]: > open source

**Comment by dominiquesydow** [comment]
Merging this PR, since all reviewers have approved of the manuscript changes:
https://github.com/openjournals/joss-reviews/issues/3951#issuecomment-996569750

I also synced the "Statement of need" between the manuscript and the documentation.

**Pull request merged**